profit/util/file_handler.py

```diff
@@ -0,0 +1,146 @@
+from profit.util.base_class import CustomABC
+from numpy import ndarray
+from typing import Union
+
+
+class FileHandler(CustomABC):
+    labels = {}
+    associated_types = {'in': 'txt', 'out': 'txt'}
+
+    @classmethod
+    def save(cls, filename, data, **kwargs):
+        """
+        Parameters:
+            filename (str)
+            data (ndarray, dict)
+            kwargs: Options like header and format for specific child classes.
+        """
+
+        ending = filename.split(".")[-1]
```

Comment:
Should we just rely on the ending to determine the filetype?
On Linux the ending can be arbitrary (even missing). Do we force the users to use the ending? What if `data.in` is actually a HDF5? For text based files there are a variety of sensible endings (`.txt`, `.csv`, `.tsv`)...

1) Using `python-magic` or calling `file` via the OS we could determine the input file type from the file header. (only on Linux though I think)

2) We could just try all filetypes until the first success.

3) We could add a config option for the filetype (also a good idea in addition to 1 or 2)

For saving only option 3 would be possible (unless the file already exists, but we don't want to overwrite it anyways?)